src/main/java/com/example/gradecalculator/controller/GradesController.java

```diff
@@ -0,0 +1,75 @@
+package com.example.gradecalculator.controller;
+
+import com.example.gradecalculator.enums.GradeTypes;
+import com.example.gradecalculator.mapper.GradeMapper;
+import com.example.gradecalculator.model.GradeTO;
+import com.example.gradecalculator.repository.UserGradeRepository;
+import com.example.gradecalculator.service.GradeService;
+import com.example.gradecalculator.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+@RestController
+@RequestMapping("/grades")
+public class GradesController {
+
+    private final GradeService gradeService;
+
+    private final UserService userService;
+    private final UserGradeRepository userGradeRepository;
+    private final GradeMapper gradeMapper;
+
+    @Autowired
+    public GradesController(GradeService gradeService, UserService userService, UserGradeRepository userGradeRepository, GradeMapper gradeMapper) {
+        this.gradeService = gradeService;
+        this.userService = userService;
+        this.userGradeRepository = userGradeRepository;
+        this.gradeMapper = gradeMapper;
+    }
+
+    @PostMapping("/save-grade")
```

Comment:
inconsisting naming comparing with other controllers. also as this is prefixed with "/grades" already it should be unneccessary to specifiy that a grade is saved
would prefer: "/grades/add"